packages/taquito/src/contract/prepare.ts

```diff
@@ -339,3 +342,35 @@ export const createSmartRollupOriginateOperation = async ({
     parameters_ty: parametersType,
   } as RPCSmartRollupOriginateOperation;
 };
+
+export const createStakingOperation = async (
+  {
+    sourceAndDestination,
+    amount,
+    fee = DEFAULT_FEE.TRANSFER,
+    gasLimit = DEFAULT_GAS_LIMIT.TRANSFER,
+    storageLimit = DEFAULT_STORAGE_LIMIT.TRANSFER,
+    mutez = false,
+  }: StakingParams,
+  entrypoint: StakingEntrypoint
+) => {
+  if (!sourceAndDestination) {
```

Comment:
This validation is not necessary, if user didn't specify source, the logic will fall into signer.publicKeyHash() default is noopSigner which will throw an error

Reply:
I still prefer to make sure this is validated here as well, not relying on logic in other parts of code that's not represented in types. (`source` is optional in `StakingParams`).
An alternative might be to define a different type with `source` required.